### PR TITLE
Move settlement updater

### PR DESCRIPTION
**Pull request by sunce86**
# Description
`OnSettlementEventUpdater` was previously refactored and ended up being dependent on `domain` and `infra` objects only. 

That means it is ready to be moved to a proper place inside `domain`.

# Changes

- [ ] Moved updater to domain::settlement::on_event module
- [ ] Updated EventIndexer to call deletion of settlement dependent data directly.

## How to test
Existing e2e tests.

### CODE REVIEW

File: crates/autopilot/src/domain/settlement/on_event.rs [hunk 1]
```diff
@@ -12,35 +12,21 @@
...
 use {
     crate::{domain::settlement, infra},
-    anyhow::{anyhow, Context, Result},
-    database::PgTransaction,
+    anyhow::{anyhow, Result},
 };
 
 #[derive(Clone)]
-pub struct OnSettlementEventUpdater {
+pub struct OnEvent {
```
Comment (MartinquaXD) [inline crates/autopilot/src/domain/settlement/on_event.rs:19]: I think the new name is not that great. What do you think about these:
* `settlement::ExecutionObserver`
* `settlement::Observer`
* `settlement::ExecutionTracker` (suggested by chatgpt)
* `settlement::Monitor`
* `settlement::PostProcessor`

I'm leaning towards `ExecutionObserver`.

**Review by MartinquaXD — COMMENTED** [review]
nit: The "to be renamed component" is the only one that creates `settlement::Settlement` instances (the `infra` is the only one consuming it). That means `settlement::Settlement::new()` can now be made private. That way we ensure that we don't have other components create these types by accident.

File: crates/autopilot/src/domain/settlement/on_event.rs (hunk 1 above)
Comment (fleupold) [reply crates/autopilot/src/domain/settlement/on_event.rs:19]: It's still a settlement, so maybe SettlementObserver (in which case you could use qualified naming for this to become `domain::settlement::Observer`)

Comment (sunce86) [reply crates/autopilot/src/domain/settlement/on_event.rs]: observer it is

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**